compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -165,7 +165,16 @@ fn parse_args<'a>(
                 let mut err = ecx.struct_span_err(p.token.span, "expected token: `,`");
                 err.span_label(p.token.span, "expected `,`");
                 p.maybe_annotate_with_ascription(&mut err, false);
-                return Err(err);
+                match token::TokenKind::Comma.similar_tokens() {
```

Comment:
This recovery is currently under the `if first` condition, but it is applicable to all commas in the list, not only the first one.

Reply:
Nice catch! I'll send a patch soon.